scripts/extensions.sh

```diff
@@ -34,64 +47,38 @@ function enable_static_extension {
     local PHP=$1
     local EXTENSION=$2
 
-    echo "Enabling ${EXTENSION} extension"
+    echo "Enabling \"${EXTENSION}\" extension"
     phpenmod -v "${PHP}" -s ALL "${EXTENSION}"
 }
 
-function enable_sqlsrv {
-    local __result=$1
-    local PHP=$2
-    local -a EXTENSIONS=("${@:3}")
-    local TO_RETURN
-
-    if [[ ! ${PHP} =~ (7.3|7.4|8.0) ]];then
-        echo "Skipping enabling of sqlsrv extension; not supported on PHP < 7.3"
-    else
-        enable_static_extension "${PHP}" sqlsrv
-    fi
-
-    TO_RETURN=$(echo "${EXTENSIONS[@]}" | sed -E -e 's/php[0-9.]+-(pdo[_-]){0,1}sqlsrv/ /g' | sed -E -e 's/\s{2,}/ /g')
-    eval "$__result='$TO_RETURN'"
-}
+PHP=$1
+declare -a EXTENSIONS=("${@:2}")
+# shellcheck disable=SC2196
+ENABLED_EXTENSIONS=$(php -m | tr '[:upper:]' '[:lower:]' | egrep -v '^[\[]' | grep -v 'zend opcache')
+EXTENSIONS_TO_INSTALL=()
 
-function enable_swoole {
-    local __result=$1
-    local PHP=$2
-    local -a EXTENSIONS=("${@:3}")
-    local TO_RETURN
+# Loop through known statically compiled/installed extensions, and enable them.
+for EXTENSION in ${EXTENSIONS[@]}; do
 
-    if [[ ! ${PHP} =~ (7.3|7.4|8.0) ]];then
-        echo "Skipping enabling of swoole extension; not supported on PHP < 7.3"
-    else
-        enable_static_extension "${PHP}" swoole
+    # Check if extension is already enabled
+    # shellcheck disable=SC2236,SC2143
+    if [ ! -z "$(echo "${ENABLED_EXTENSIONS}" | grep "${EXTENSION}")" ]; then
```

Comment:
I'd love to figure out a way to do this without needing to suppress the shellcheck warnings, but I tried a bunch of variations, none of which worked as intended.

Reply:
I really hate using `echo` to search for stuff but I havent figured out a better way. My bash scripting skills are a bit rusty.

Reply:
Actually, there IS a way to do this without pipe + grep, since what we're trying to determine is simply whether or not `$ENABLED_EXTENSIONS` contains `$EXTENSION`:

```bash
if [[ "${ENABLED_EXTENSIONS}" =~ "${EXTENSION}" ]]; then
```

Reply:
Yah, but sadly, this does not support "words" (and this is not achieved with `grep` as well).
So this might lead to problems when searching for exact matches.

Do you have an idea how to solve that?

Reply:
Okay, I managed to create a boundary expression.